h/activity/query.py

```diff
@@ -123,11 +123,12 @@ def execute(request, query, page_size):
     # Add group information to buckets and present annotations
     for timeframe in result.timeframes:
         for bucket in timeframe.document_buckets.values():
+            bucket.presented_annotations = []
             for index, annotation in enumerate(bucket.annotations):
-                bucket.annotations[index] = {
+                bucket.presented_annotations.append({
```

Comment:
Oh yeah, it was modifying the list that it was iterating over! I agree with your change